airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -876,6 +877,61 @@ def get_task_instance_count(
     return count or 0
 
 
+@router.get("/previous/{dag_id}/{task_id}", status_code=status.HTTP_200_OK)
+def get_previous_task_instance(
+    dag_id: str,
+    task_id: str,
+    session: SessionDep,
+    logical_date: Annotated[UtcDateTime | None, Query()] = None,
+    run_id: Annotated[str | None, Query()] = None,
```

Comment:
A general critique to this feature (which is carried over from Airflow 2) rather than the PR… why is a `run_id` filter useful? If you know the run ID, why not just get that particular task instance directly?

Reply:
True, will remove